src/views/CreatePage.vue

```diff
@@ -197,38 +198,30 @@ onUnmounted(() => {
     changeModeByShortCut(e)
     handleKeyDownSelectedNodeDelete(e)
   })
-
-  // canvasHistoryのリセット
-  historyStep.value = 0
-  canvasHistory.value = [{ lines: [], texts: [], images: [] }]
 })
 
+// セーブ中は他のページに遷移できない
 router.beforeEach(() => {
   if (saveState.value === 'loading') {
     return false
   }
   return true
 })
 
+// キャンバスの保存
 const saveCanvas = async (): Promise<void> => {
   // 選択を解除
-  selectedShapeId.value = ''
-  configShapeTransformer.value.nodes = []
-
+  useStoreTransformer().$reset()
   // btn loadingに変更
   saveState.value = 'loading'
 
   // 途中からの場合
-  const canvasVal = canvasId.value
-  if (
-    typeof canvasVal === 'string' &&
-    canvases.value[canvasVal] !== undefined
-  ) {
+  if (canvases.value[canvasId.value] !== undefined) {
     // createdAtがある場合
-    if (canvases.value[canvasVal].createdAt !== undefined) {
+    if (canvases.value[canvasId.value].createdAt !== undefined) {
       // isShareがある場合
-      if (canvases.value[canvasVal].isShare !== undefined) {
-        await updateDoc(doc(db, 'canvas', canvasVal), {
+      if (canvases.value[canvasId.value].isShare !== undefined) {
+        await updateDoc(doc(db, 'canvas', canvasId.value), {
```

Comment:
処理一つにまとめて良いと思います。
処理速度はそこまで変わらないので、可読性を優先した方が良いと思います。

```suggestion
       await updateDoc(doc(db, 'canvas', canvasId.value), {
        name: inputText.text === '' ? 'タイトル' : inputText.text,
        lines: lines.value,
        texts: texts.value,
        konvaImages: changeKonvaImagesToFirestoreCanvasImages(
          konvaImages.value,
        ),
        createdAt: canvases.value[canvasId.value].createdAt === undefined
          ? Timestamp.now()
          : canvases.value[canvasId.value].createdAt,
        updatedAt: Timestamp.now(),
        isShare:
          canvases.value[canvasId.value].isShare === undefined
            ? false
            : canvases.value[canvasId.value].isShare,
      })
```

Reply:
そっか、三項演算子使えばよかったんですね…
プロパティ割り当てる時にも使えるのか、全然頭になかったです笑
ありがとうございます。